core/modules/replica/IngestServerConnection.cc

```diff
@@ -77,9 +77,11 @@ bool isErrorCode(boost::system::error_code const& ec,
 bool readIntoBuffer(boost::asio::ip::tcp::socket& socket,
                     shared_ptr<ProtocolBuffer> const& ptr,
                     size_t bytes) {
+    // Make sure the buffer has enough space to accommodate the data
+    // of the message. Note, this call may throw an exception which is
+    // supposed to be caught by the method's caller.
+    ptr->resize(bytes);
```

Comment:
again; catch? Or are you saying it should be caught by *this* method's caller?

Reply:
Exactly! This method may throw exceptions. The callers are supposed to call them.